src/messages.rs

```diff
@@ -287,22 +352,12 @@ impl RoutingMessage {
         self.content.priority()
     }
 
-    /// Replaces this message's contents with its hash.
-    pub fn to_grp_msg_hash(&self) -> Result<RoutingMessage, RoutingError> {
-        let content = match self.content {
-            MessageContent::GetNodeNameResponse { .. } |
-            MessageContent::GetCloseGroupResponse { .. } |
-            MessageContent::UserMessagePart { .. } => {
-                let serialised_msg = try!(serialise(self));
-                MessageContent::GroupMessageHash(sha256::hash(&serialised_msg), self.priority())
-            }
-            _ => self.content.clone(),
-        };
-        Ok(RoutingMessage {
-            src: self.src,
-            dst: self.dst,
-            content: content,
-        })
+    /// Returns a `DirectMessage::MessageSignature` for this message.
+    pub fn to_signature(&self, sign_key: &sign::SecretKey) -> Result<DirectMessage, RoutingError> {
```

Comment:
I guess since "sign" is a verb, I'd have opted for `secret_key` or `signing_key` here, but not a big deal.